test/scpi.py

```diff
@@ -0,0 +1,140 @@
+import time
+
+import iio
+
+import adi
+import numpy as np
+import paramiko
+import pyvisa
+from pyvisa import constants
+
+supported_instruments = ["HAMEG Instruments,HM8123,5.12"]
+
+
+def find_instrument():
+    rm = pyvisa.ResourceManager()
+    all_resources = rm.list_resources()
+    for i in range(0, len(all_resources)):
+        my_instrument = rm.open_resource(
+            all_resources[i],
+            baud_rate=9600,
+            data_bits=8,
+            parity=constants.Parity.none,
+            stop_bits=constants.StopBits.one,
+            write_termination="\r",
+            read_termination="\r",
+        )
+
+        try:
+            idn = my_instrument.query("*IDN?")
+            if idn in supported_instruments:
+                break
+            else:
+                print("This instrument is currently not supported!")
+        except Exception as e:
+            my_instrument = None
+            continue
+    return my_instrument
+
+
+def select_channel(instr):
+    idn = instr.query("*IDN?")
+    if idn == "HAMEG Instruments,HM8123,5.12":
+        ch = ["FRA", "FRB", "FRC"]
+        for i in range(0, len(ch)):
+            instr.write(ch[i])
+            time.sleep(3)
+            xmt = instr.query("XMT?")
+            time.sleep(1)
+            if xmt != "Not Available":
+                return xmt
+
+
+def get_freq(instr):
+    # set gate time to 600 for better precision
+    instr.write("SMT600")
+    xmt = select_channel(instr)
+    frq_str = xmt.split(" ", 2)
+    scale = 1
+    if frq_str[1] == "GHz":
+        scale = pow(10, 9)
+    elif frq_str[1] == "MHz":
+        scale = pow(10, 6)
+    elif frq_str[1] == "KHz":
+        scale = pow(10, 3)
+    return float(frq_str[0]) * scale
+
+
+def get_clk_rate(classname, iio_uri):
+    sdr = eval(classname + "(uri='" + iio_uri + "')")
+    sdr._ctrl.debug_attrs["adi,clk-output-mode-select"].value = "1"
+    sdr._ctrl.debug_attrs["initialize"].value = "1"
+
+    full_uri = iio_uri.split(":", 2)
+    if full_uri[0] != "ip":
+        print("Tuning currently supported only for ip URIs")
+    ip = full_uri[1]
+
+    ssh_client = paramiko.SSHClient()
+    ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh_client.connect(ip, username="root", password="analog")
+    ssh_stdin, ssh_stdout, ssh_stderr = ssh_client.exec_command(
+        "cat /sys/kernel/debug/clk/ad9361_ext_refclk/clk_rate"
```

Comment:
Can we use IIO to get this value? If not can we add a check to the SSH stderr to make sure we know if that breaks

Reply:
I added a check, is simply printing the stderr in order?

Reply:
Is there an error code we can check here? Like if not 0 raise Exception